### PR TITLE
Fix flakiness of BlameDataCollectorShouldOutputDumpFile acceptance test

### PR DESCRIPTION
**Pull request by karanjitsingh**
This PR fixes the occasional failing of BlameDataCollectorShouldOutputDumpFile acceptance test
This PR also changes BlameCollector's LogErrors to LogWarnings

### CODE REVIEW

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs [hunk 1]
```diff
@@ -312,9 +314,12 @@ private void AddNewFileTransfer(FileTransferInformation fileTransferInfo, AsyncC
                     {
                         if (t.Exception == null)
                         {
-                        // Uri doesn't recognize file paths in unix. See https://github.com/dotnet/corefx/issues/1745
-                        var attachmentUri = new UriBuilder() { Scheme = "file", Host = "", Path = localFilePath }.Uri;
-                            this.AttachmentSets[fileTransferInfo.Context][uri].Attachments.Add(new UriDataAttachment(attachmentUri, fileTransferInfo.Description));
+                            lock (lockObject)
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs]: Try to use `ConcurrentDictionary` here.

Comment (karanjitsingh) [reply src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs]: Switched to concurrent dictionaries wherever required but cannot remove lock as a lot of code depends on IList

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs [hunk 2]
```diff
@@ -71,8 +74,8 @@ protected DataCollectionAttachmentManager(IFileHelper fileHelper)
         {
             this.fileHelper = fileHelper;
             this.cancellationTokenSource = new CancellationTokenSource();
-            this.attachmentTasks = new Dictionary<DataCollectionContext, List<Task>>();
-            this.AttachmentSets = new Dictionary<DataCollectionContext, Dictionary<Uri, AttachmentSet>>();
+            this.attachmentTasks = new ConcurrentDictionary<DataCollectionContext, List<Task>>();
+            this.AttachmentSets = new ConcurrentDictionary<DataCollectionContext, IDictionary<Uri, AttachmentSet>>();
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs]: Add tests for this change.

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs [hunk 3]
```diff
@@ -312,9 +315,12 @@ private void AddNewFileTransfer(FileTransferInformation fileTransferInfo, AsyncC
...
                         if (t.Exception == null)
                         {
-                        // Uri doesn't recognize file paths in unix. See https://github.com/dotnet/corefx/issues/1745
-                        var attachmentUri = new UriBuilder() { Scheme = "file", Host = "", Path = localFilePath }.Uri;
-                            this.AttachmentSets[fileTransferInfo.Context][uri].Attachments.Add(new UriDataAttachment(attachmentUri, fileTransferInfo.Description));
+                            lock (lockObject)
+                            {
+                                // Uri doesn't recognize file paths in unix. See https://github.com/dotnet/corefx/issues/1745
+                                var attachmentUri = new UriBuilder() { Scheme = "file", Host = "", Path = localFilePath }.Uri;
+                                this.AttachmentSets[fileTransferInfo.Context][uri].Attachments.Add(new UriDataAttachment(attachmentUri, fileTransferInfo.Description));
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs]: Add lock to only list add method.

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs [hunk 4]
```diff
@@ -24,6 +25,8 @@ namespace Microsoft.VisualStudio.TestPlatform.Common.DataCollector
     /// </summary>
     internal class DataCollectionAttachmentManager : IDataCollectionAttachmentManager
     {
+        private static object lockObject = new object();
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs]: rename this to `attachmentTaskLock`.

**Review by smadala — APPROVED** [review]

**Comment by karanjitsingh** [comment]
@dotnet-bot test Windows_NT / Debug Build

**Pull request merged**